app/src/components/explorer/Pagination.vue

```diff
@@ -1,6 +1,17 @@
 <template>
   <div class="viz-pagination-container">
-    <span>Page {{ cpage }} of {{ tpages }}</span>
+    <span class="viz-pagination u_centralize_content">
+      <div class="pagination-field">
+         Page
+      </div>
+      <md-field :class="pgInputClass" class="u_width--small">
+        <md-input id="pageInput" v-model="pageInput" @change="goToPage($event.target.value)" required></md-input>
+        <span class="md-error">Out of range</span>
+      </md-field>
```

Comment:
Should we be allowing users to input page numbers here, I'll suggest taking this input to the top of the page so it sits beside page size.

Reply:
Sure, I can move it. The main reason I put it there was because I thought it made sense visually, but I can see how it might cause issues